payjoin/tests/integration.rs

```diff
@@ -55,16 +65,28 @@ mod integration {
             // **********************
             // Inside the Receiver:
             // this data would transit from one party to another over the network in production
-            let response = handle_v1_pj_request(req, headers, receiver);
+            let response = handle_v1_pj_request(req, headers, &receiver);
             // this response would be returned as http response to the sender
 
             // **********************
             // Inside the Sender:
-
             // Sender checks, signs, finalizes, extracts, and broadcasts
             let checked_payjoin_proposal_psbt = ctx.process_response(&mut response.as_bytes())?;
             let payjoin_tx = extract_pj_tx(&sender, checked_payjoin_proposal_psbt)?;
             sender.send_raw_transaction(&payjoin_tx)?;
+
+            // Check resulting transaction and balances
+            let input_count = payjoin_tx.input.len() as u64;
+            let output_count = payjoin_tx.output.len() as u64;
+            let tx_weight = BASE_WEIGHT + input_count * INPUT_WEIGHT + output_count * OUTPUT_WEIGHT;
+            let network_fees = tx_weight * FeeRate::BROADCAST_MIN;
+            assert_eq!(input_count, 2);
+            assert_eq!(output_count, 2);
```

Comment:
> I'm a bit confused also why manual calculation is done rather than using [Transaction::weight()](https://docs.rs/bitcoin/latest/bitcoin/blockdata/transaction/struct.Transaction.html#method.weight)

Simple answer is that I didn't think of that 🤦 That does seem like a better approach though if we can get it to pass.